my_enumerable.rb

```diff
@@ -0,0 +1,210 @@
+
+# rubocop:disable all
+module Enumerable
+  def my_each
+    for item in self
+      if block_given?
+        yield(item)
+      else
+        item
+      end
+    end
+  end
```

Comment:
- [ ] my_each

-  returns an Enumerator if no block is given(just like the orginal `each`):

`p [1, 2, 3].my_each` return `#<Enumerator>`

This can be achieved with [to_enum](https://apidock.com/ruby/Object/to_enum)